### PR TITLE
Submission removed observer

### PR DESCRIPTION
**Pull request by tuanngocnguyen**
This  adds an observer to handle submission removal event, follow up the suggestion (number 2) by Tomo on

https://github.com/turnitin/moodle-plagiarism_turnitin/issues/556

### CODE REVIEW

**Comment by jack-tii** [comment]
@tuanngocnguyen Thanks for your contribution! Could you rebase against the develop branch?

File: version.php [hunk 1]
```diff
@@ -19,7 +19,7 @@
  * @copyright 2012 iParadigms LLC
  */
 
-$plugin->version = 2023033001;
+$plugin->version = 2023033002;
```
Comment (jack-tii) [inline version.php:22]: We update the version number based on our the date of release, we would need to revert this change befor this could be merged

**Comment by tuanngocnguyen** [comment]
Hi @jack-tii ,

Thanks for your note,

I have open a new PR with a different branch instead of re-basing the current branch (it may be being used)

Hence I am closing this PR

The new PR is https://github.com/turnitin/moodle-plagiarism_turnitin/pull/730
